pkg/kinflate/configmap_secret.go

```diff
@@ -18,74 +18,114 @@ package kinflate
 		return fmt.Errorf("cannot use a duplicate name %q for %s", oldName, gvk)
 	}
 	accessor.SetName(newName)
-	m[gvkn] = newNameObject{
-		newName: newName,
-		obj:     obj,
-	}
+	m[gvkn] = obj
 	return nil
 }
 
-func populateMapOfConfigMapAndSecret(r *resource, m map[groupVersionKindName]newNameObject) error {
-	for _, cm := range r.configmaps {
-		corev1cm, err := constructConfigMap(cm)
+func populateConfigMapAndSecretMap(manifest *manifest.Manifest, m map[kutil.GroupVersionKindName]*unstructured.Unstructured) error {
+	for _, cm := range manifest.Configmaps {
+		unstructuredConfigMap, nameWithHash, err := constructConfigmapAndGenerateName(cm)
 		if err != nil {
 			return err
 		}
-		h, err := hash.ConfigMapHash(corev1cm)
-		if err != nil {
-			return err
-		}
-		nameWithHash := fmt.Sprintf("%s-%s", corev1cm.Name, h)
-		err = populateMap(m, corev1cm, nameWithHash)
+		err = populateMap(m, unstructuredConfigMap, nameWithHash)
 		if err != nil {
 			return err
 		}
 	}
 
-	for _, secret := range r.secrets {
-		corev1secret, err := constructSecret(secret)
-		if err != nil {
-			return err
-		}
-		h, err := hash.SecretHash(corev1secret)
+	for _, secret := range manifest.Secrets {
+		unstructuredSecret, nameWithHash, err := constructSecretAndGenerateName(secret)
 		if err != nil {
 			return err
 		}
-		nameWithHash := fmt.Sprintf("%s-%s", corev1secret.Name, h)
-		err = populateMap(m, corev1secret, nameWithHash)
+		err = populateMap(m, unstructuredSecret, nameWithHash)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func constructConfigmapAndGenerateName(cm manifest.ConfigMap) (*unstructured.Unstructured, string, error) {
+	corev1CM, err := constructConfigMap(cm)
+	if err != nil {
+		return nil, "", err
+	}
+	h, err := hash.ConfigMapHash(corev1CM)
+	if err != nil {
+		return nil, "", err
+	}
+	nameWithHash := fmt.Sprintf("%s-%s", corev1CM.GetName(), h)
+	unstructuredCM, err := v1ConfigMapToUnstructured(corev1CM)
+	return unstructuredCM, nameWithHash, err
+}
+
+func constructSecretAndGenerateName(secret manifest.Secret) (*unstructured.Unstructured, string, error) {
+	corev1Secret, err := constructSecret(secret)
+	if err != nil {
+		return nil, "", err
+	}
+	h, err := hash.SecretHash(corev1Secret)
+	if err != nil {
+		return nil, "", err
+	}
+	nameWithHash := fmt.Sprintf("%s-%s", corev1Secret.GetName(), h)
+	unstructuredCM := v1SecretToUnstructured(corev1Secret)
+	return unstructuredCM, nameWithHash, nil
+}
+
+func v1ConfigMapToUnstructured(in *corev1.ConfigMap) (*unstructured.Unstructured, error) {
+	marshaled, err := json.Marshal(in)
+	if err != nil {
+		return nil, err
+	}
+	var out unstructured.Unstructured
+	err = out.UnmarshalJSON(marshaled)
+	return &out, err
+}
+
+// We cannot marshal a corev1 Secret and then unmarshal it to a unstructured.
+// Because there are []byte fields (Data is map[string][]byte) in corev1 Secret.
+// In goland, unmarshal an marshaled byte array will become a base64 encoded string.
```

Comment:
goland?  you mean the IDE?   don't understand, please reword.

Reply:
Typo: it should be `golang`.

Reply:
To be more concrete, here is an example: https://play.golang.org/p/zcW07XETcsB

Reply:
Will remove this function. We don't really need to worry about it.